lib/galaxy/webapps/galaxy/api/jobs.py

```diff
@@ -257,148 +266,262 @@ def common_problems(self, trans: ProvidesUserContext, id, **kwd):
+        """
+        tool_id = payload.tool_id
 
-            This API endpoint is unstable and tied heavily to Galaxy's JS client code,
-            this endpoint will change frequently.
+        tool = trans.app.toolbox.get_tool(tool_id)
+        if tool is None:
+            raise exceptions.ObjectNotFound("Requested tool not found")
+        # TODO the inputs are actually a dict, but are passed as a JSON dump
+        # maybe change it?
+        inputs = json.loads(payload.inputs)
+        # Find files coming in as multipart file data and add to inputs.
+        for k, v in payload.__annotations__.items():
+            if k.startswith("files_") or k.startswith("__files_"):
+                inputs[k] = v
+        request_context = WorkRequestContext(app=trans.app, user=trans.user, history=trans.history)
+        all_params, all_errors, _, _ = tool.expand_incoming(
+            trans=trans, incoming=inputs, request_context=request_context
+        )
+        if any(all_errors):
+            return []
+        params_dump = [tool.params_to_strings(param, trans.app, nested=True) for param in all_params]
+        jobs = []
+        for param_dump, param in zip(params_dump, all_params):
+            job = self.service.job_search.by_tool_input(
+                trans=trans,
+                tool_id=tool_id,
+                tool_version=tool.version,
+                param=param,
+                param_dump=param_dump,
+                job_state=payload.state,
+            )
+            if job:
+                jobs.append(job)
+        return [EncodedJobDetails(**single_job.to_dict("element")) for single_job in jobs]
 
-        :type   job_id: string
-        :param  job_id: Encoded job id
+    @router.get(
+        "/api/jobs/{id}",
+        name="show_job",
+        summary="Return dictionary containing description of job data.",
+    )
+    def show(
+        self,
+        id: Annotated[DecodedDatabaseIdField, JobIdPathParam],
+        full: Annotated[Optional[bool], FullShowQueryParam] = False,
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> Dict[str, Any]:
+        return self.service.show(trans, id, bool(full))
 
-        :type   dataset_id: string
-        :param  dataset_id: Encoded HDA or LDDA id
+    # TODO find the mapping of the legacy route
+    # TODO rename? --> function cancels/stops job (no deletion involved?)
+    @router.delete(
+        "/api/jobs/{id}",
+        name="cancel_job",
+        summary="Cancels specified job",
+    )
+    def delete(
+        self,
+        id: Annotated[DecodedDatabaseIdField, JobIdPathParam],
+        trans: ProvidesUserContext = DependsOnTrans,
+        payload: Annotated[Optional[DeleteJobPayload], DeleteJobBody] = None,
+    ) -> bool:
+        job = self.service.get_job(trans=trans, job_id=id)
+        if payload:
+            message = payload.message
+        else:
+            message = None
+        return self.service.job_manager.stop(job, message=message)
```

Comment:
This endpoint actually cancels a job, while this may lead to deletion, I think naming the endpoint delete is not really intuitive. I took the freedom to assign its name="cancel_job", but I do not want to rename the endpoint yet. Any suggestions?

Reply:
Thanks, this is fine.

Reply:
So I can rename it?

Reply:
No, i meant this is fine as is. We can discuss this outside of the PR context ... there's a mode in the equivalent old-style controller that also deletes job outputs.

Reply:
Ok thanks :)